main.py

```diff
@@ -66,42 +68,65 @@ async def marco(ctx):
 
     @bot.command()
     async def membercardsranked(ctx, *args):
-        invalid_cmd_reply = 'Please provide a clan tag. E.g.\n`!membercardsranked 9GULPJ9L`'
+        card_type_arg = 'all'
         if len(args) == 0:
+            invalid_cmd_reply = 'Please provide a clan tag. E.g.\n`!membercardsranked 9GULPJ9L`'
             await ctx.send(invalid_cmd_reply)
             return
+        elif len(args) == 2:
+            card_type_arg = args[1].lower().rstrip('s')
+            check_result = await card_type_check(ctx, card_type_arg)
+            if check_result is not True:
+                return
         clan_tag = args[0]
 
-        async with ctx.typing():
-            # TODO: might want some try catch before sending result in case API requests fail or something...
-            print("Started fetch ranked members processing...")
-            clan_info, clan_members_ranked = clan_members_ranker.get_clan_cards_rank(clan_tag)
-            print("Completed fetch ranked members processing")
+        await fetch_ranked_members(ctx, clan_tag, card_type_arg)
 
-        await ctx.send(embed=create_clan_members_ranked_embed(clan_info, clan_members_ranked))
+    @ bot.command()
+    async def ausclan(ctx, *args):
+        card_type_arg = 'all'
+        if len(args) == 1:
+            card_type_arg = args[0].lower().rstrip('s')
```

Comment:
You could do the .lower.rstip('s') in the function itself by passing it card_type_check(ctx, args[0])

Reply:
as discussed, leave it there